### PR TITLE
ALP for A4A

### PR DESCRIPTION
**Pull request by zhouyx**
Work for #5212 

- Tested with a fake A4A. I failed to get a working A4A ad that navigates to a valid amp page. So I tested that we can get the redirect url correctly from an A4A ad. Then I mocked a fake url in our code base and tested that we get the desired redirect behavior. I think these two tests get the redirect process fully covered, we should be good now with an experiment flag.

- Another thing @cramforce mentioned, and I think we need to finalize is what to do with `target = _blank` in the case of A2A redirect. This is a preference thing, I would vote for open in new window since user specifically ask for `target=_blank`.  

cc @jasti

### CODE REVIEW

File: examples/viewer.html [hunk 1]
```diff
@@ -504,7 +504,7 @@
       new Viewer(
           'container1',
           '1',
-          './everything.amp.max.html',
+          './a4a.amp.max.html?exp=a4a:-1&force_a4a_ctypes=1,2',
```
Comment (zhouyx) [inline examples/viewer.html]: i will revert this

**Comment by zhouyx** [comment]
PTAL

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 2]
```diff
@@ -804,4 +807,15 @@ export class AmpA4A extends AMP.BaseElement {
        metaData.cssUtf16CharOffsets[0],
        metaData.cssUtf16CharOffsets[1]);
  }
+
+  registerAlpHandler_(iframeWin) {
+    if (!isExperimentOn(this.win, 'alp-for-a4a')) {
+      return;
+    }
+    iframeWin.addEventListener('click', event => {
```
Comment (cramforce) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Does this work? I think `iframeWin.document.documentElement` would be more idiomatic.

Comment (zhouyx) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Ha! changed to register listener on element.

**Pull request merged**